lxd/seccomp/seccomp.go

```diff
@@ -1856,6 +1885,188 @@ func (s *Server) HandleSysinfoSyscall(c Instance, siov *Iovec) int {
 	return 0
 }
 
+type nullWriteCloser struct {
+	*bytes.Buffer
+}
+
+// Close is a no-op closer implementation
+func (nwc *nullWriteCloser) Close() error {
+	return nil
+}
+
+// HandleFinitModuleSyscall handles finit_module syscalls.
+func (s *Server) HandleFinitModuleSyscall(c Instance, siov *Iovec) int {
+	ctx := logger.Ctx{"container": c.Name(),
+		"project":               c.Project().Name,
+		"syscall_number":        siov.req.data.nr,
+		"audit_architecture":    siov.req.data.arch,
+		"seccomp_notify_id":     siov.req.id,
+		"seccomp_notify_flags":  siov.req.flags,
+		"seccomp_notify_pid":    siov.req.pid,
+		"seccomp_notify_fd":     siov.notifyFd,
+		"seccomp_notify_mem_fd": siov.memFd,
+	}
+
+	defer logger.Debug("Handling finit_module syscall", ctx)
+
+	// int fd
+	fd := C.int(siov.req.data.args[0])
+
+	// const char *param_values
+	cBuf := [4096]C.char{}
+	_, err := C.pread(C.int(siov.memFd), unsafe.Pointer(&cBuf[0]), C.size_t(4096), C.off_t(siov.req.data.args[1]))
+	if err != nil {
+		ctx["err"] = fmt.Sprintf("Failed to read memory for finit_module syscall: %s", err)
+		if s.s.OS.SeccompListenerContinue {
+			ctx["syscall_continue"] = "true"
+			C.seccomp_notify_update_response(siov.resp, 0, C.uint32_t(seccompUserNotifFlagContinue))
+			return 0
+		}
+
+		return int(-C.EFAULT)
+	}
+
+	paramValues := C.GoString(&cBuf[0])
+
+	// disallow specifying module parameters, it's just not safe
```

Comment:
An expanded comment here explaining *why* would be helpful to others.

Reply:
Done!